apps/amiapp_flutter/ios/Runner/AppDelegate.swift

```diff
@@ -38,6 +40,39 @@ import FirebaseCore
     override func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
         MessagingPush.shared.application(application, didFailToRegisterForRemoteNotificationsWithError: error)
     }
+    
+    // Function called when a push notification is clicked or swiped away.
+    override func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
```

Comment:
We don't have the support to observe these callbacks in the `flutter/dart` file? similar to what we are doing in react native?

Reply:
Good question. There probably is a way, but I was not able to find a way to do it. I am guessing it will require a flutter plugin to do that? From searching, I have not been able to find one. Happy to change it if someone finds a resource.

Reply:
You have already added the plugin "flutter_local_notifications", [here](https://blog.codemagic.io/flutter-local-notifications/) is something I found.

Reply:
Oh, I forgot to mention that I did try to install a callback for `flutter_local_notifications` but was not able to successfully get it to work. I installed the callback functions according to the official docs, I noticed that the CIO SDK successfully installed the native push event handler for the plugin, but my callback function never got executed. 

Rather then dig deeper into it, I decided to go with this approach since I knew it would work. 

I have not found the docs for the plugin to be very helpful so it would take me a bit of time to debug this and see why the callback function is not being called.

Reply:
Got Dart callbacks working with `flutter_local_notifications` and pushed the commit to this PR. 

I figured out the problem. I was overriding the native push event callback functions in the `AppDelegate` which then prevented `flutter_local_notifications` from ever getting called. 

You'll now see in the PR that there is only 1 line required in `AppDelegate` to get this to work.